crates/jstz_core/src/native.rs

```diff
@@ -455,13 +447,25 @@ pub trait NativeClass {
     /// The attributes the class will be bound with, default is `writable`, `enumerable`, `configurable`.
     const ATTRIBUTES: Attribute = Attribute::all();
 
-    /// The constructor of the class.
-    fn constructor(
-        this: &JsNativeObject<Self::Instance>,
+    /// Creates the internal data for an instance of this class.
+    fn data_constructor(
+        target: &JsValue,
         args: &[JsValue],
         context: &mut Context<'_>,
     ) -> JsResult<Self::Instance>;
 
+    /// Initializes the properties of the constructed object for an instance of this class.
+    ///
+    /// Useful to initial additional properties for the construced object that aren't part of the
+    /// internal data are rely on the arguments or this.
```

Comment:
this comment could be improved

Reply:
In what sense? What do you not understand from the comment?

Reply:
The whole sentence doesn't make much sense with the "data are rely".
There is also a grammar issue - construced->constructed.
Overall it is also slightly confusing and would benefit from simplification clearly separating what is meant for rust and what for Javascript and why we need two constructors, possibly with a small example of how the are used.

Reply:
Fixed typos -- linked example usage in `Url::object_constructor`.

Reply:
We have 3 kinds of constructors:
- The `raw` constructor
- The `data` constructor
- The `object` constructor

The `raw` constructor is the `NativeFunction` that is called when we want to construct an object e.g. the function called when evaluating `new URL(...)`. 

The `data` constructor is the constructor for creating the Rust value (of type `T`) that is required to construct the `JsNativeObject<T>` -- the JavaScript representation of `T`. 

The `object` constructor is the constructor that has access to `this` and `args`; it is called after the `data` constructor since it requires `this : &JsNativeObject<T>`. This constructor is useful for setting properties that rely on `this` (e.g. the required cycle between `URL` and `URLSearchParams`)